src/Paket.Core/Dependencies/NuGetV2.fs

```diff
@@ -20,47 +20,57 @@ open Paket.Requirements
-            | Some node -> node
-            | None -> failwithf "unable to parse data from %s" url
-
-        let readEntryVersion = Some
-                               >> optGetNode "properties"
-                               >> optGetNode "Version"
-                               >> Option.map (fun node -> node.InnerText)
-
-        let entriesVersions = feed |> getNodes "entry" |> List.choose readEntryVersion
-
-        let! linksVersions =
-            feed
-            |> getNodes "link"
-            |> List.filter (fun node -> node |> getAttribute "rel" = Some "next")
-            |> List.choose (getAttribute "href")
-            |> List.filter (fun x -> x <> url)
-            |> List.map (followODataLink auth)
-            |> Async.Parallel
-
-        return
-            linksVersions
-            |> Seq.collect id
-            |> Seq.append entriesVersions
-            |> Seq.toArray
-    }
+let private followODataLink auth url =
+    let rec followODataLinkSafe (knownVersions:Set<_>) url =
+        async {
+            let! raw = getFromUrl(auth, url, acceptXml)
+            if String.IsNullOrWhiteSpace raw then return [||] else
+            let doc = XmlDocument()
+            doc.LoadXml raw
+            let feed =
+                match doc |> getNode "feed" with
+                | Some node -> node
+                | None -> failwithf "unable to parse data from %s" url
+
+            let readEntryVersion = Some
+                                   >> optGetNode "properties"
+                                   >> optGetNode "Version"
+                                   >> Option.map (fun node -> node.InnerText)
+
+            let entriesVersions = feed |> getNodes "entry" |> List.choose readEntryVersion
+            let newSet = entriesVersions |> Set.ofList
+            let noNewFound = newSet.IsSubsetOf knownVersions
+            if noNewFound then
+                traceWarnfn "Feed returned a next link but no new versions were found. Either the feed is broken or paket sends the wrong requests (noticed on '%O')" url
+
+            let! linksVersions =
+                async {
+                    if noNewFound then
+                        return [||]
+                    else
+                        return!
+                            feed
+                            |> getNodes "link"
+                            |> List.filter (fun node -> node |> getAttribute "rel" = Some "next")
+                            |> List.choose (getAttribute "href")
+                            |> List.filter (fun x -> x <> url)
+                            |> List.map (followODataLinkSafe (Set.union newSet knownVersions))
+                            |> Async.Parallel
+                }
+
+            return
+                linksVersions
+                |> Seq.collect id
+                |> Seq.append entriesVersions
+                |> Seq.toArray
+        }
+    followODataLinkSafe Set.empty url
```

Comment:
This part needs to be reviewed. Is there some paket bug I'm missing or is this workaround OK?
/cc @forki @cloudRoutine see discussion in https://github.com/fsprojects/Paket/issues/2435